### PR TITLE
magento/devdocs#: Extends information about creating a dynamic row system config.

### PR DESCRIPTION
**Pull request by atwixfirster**
## Purpose of this pull request

This pull request (PR) extends current [Creating a dynamic row system config](https://devdocs.magento.com/guides/v2.2/ext-best-practices/tutorials/dynamic-row-system-config.html) topic and adds information how to add a custom drop-down option.

dynamic-rows-config-result

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.2/ext-best-practices/tutorials/dynamic-row-system-config.html
- https://devdocs.magento.com/guides/v2.3/ext-best-practices/tutorials/dynamic-row-system-config.html

whatsnew
Added additional steps and code in [Creating a dynamic row system config](https://devdocs.magento.com/guides/v2.3/ext-best-practices/tutorials/dynamic-row-system-config.html).

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/ext-best-practices/tutorials/dynamic-row-system-config.md [hunk 1]
```diff
@@ -56,14 +64,113 @@ class Ranges extends AbstractFieldArray
...
+    {
+        return $this->setId($value);
+    }
+
+    /**
+     * Render block HTML
+     *
+     * @return string
+     */
+    public function _toHtml(): string
```
Comment (rogyar) [inline guides/v2.2/ext-best-practices/tutorials/dynamic-row-system-config.md]: It's not a big deal, but should we override the parent method scope of `_toHtml()`? It's `protected` in the parent class.

Comment (atwixfirster) [reply guides/v2.2/ext-best-practices/tutorials/dynamic-row-system-config.md]: @rogyar , I see such approach in the existing Magento classes: 
- [Magento\CatalogInventory\Block\Adminhtml\Form\Field\Customergroup::_toHtml()](https://github.com/magento/magento2/blob/2.3.3/app/code/Magento/CatalogInventory/Block/Adminhtml/Form/Field/Customergroup.php#L106)
- [Magento\Braintree\Block\Adminhtml\Form\Field\Countries::_toHtml()](https://github.com/magento/magento2/blob/2.3.3/app/code/Magento/Braintree/Block/Adminhtml/Form/Field/Countries.php#L40)
- [Magento\Braintree\Block\Adminhtml\Form\Field\CcTypes::_toHtml()](https://github.com/magento/magento2/blob/2.3.3/app/code/Magento/Braintree/Block/Adminhtml/Form/Field/CcTypes.php#L43)

So, in my PR I've followed the same logic.

Thanks.

Comment (rogyar) [reply guides/v2.2/ext-best-practices/tutorials/dynamic-row-system-config.md]: I believe it's a sort of mechanical mistake. But, not a big deal, as I said.

**Comment by dobooth** [comment]
running tests

**Pull request merged**